Dockerfile

```diff
@@ -82,6 +98,10 @@ ENV CDM_SHARED_DIR=/cdm_shared_workspace
 RUN mkdir -p ${CDM_SHARED_DIR} && chmod -R 777 ${CDM_SHARED_DIR}
 RUN chown -R spark_user:spark $CDM_SHARED_DIR
 
+# Allow spark_user to use sudo without a password
+# TODO: use `sudospawner` in JupyterHub to avoid this (https://jupyterhub.readthedocs.io/en/stable/howto/configuration/config-sudo.html)
+RUN echo "spark_user ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
```

Comment:
How do you ensure users can't access this account?

Reply:
users logged in as individual system user. I don't know they can have access to spark_user.

Reply:
So you have to make a user account for each user?

Reply:
Yes. Jupyterhub operates only on system users.

Reply:
Ok, so what the spawner is going to do it make a new user if the user & set up home if it doesn't exist, and rely on the linux permissions to make sure they can't read other user's envs?

Reply:
Yea. Will implement those in the future PRs.

Reply:
:+1: